graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/RemoteReindexAllowlist.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.storage.opensearch2;
+
+import jakarta.annotation.Nonnull;
+import org.apache.commons.lang3.StringUtils;
+import org.graylog.shaded.opensearch2.org.apache.lucene.util.automaton.Automaton;
+import org.graylog.shaded.opensearch2.org.apache.lucene.util.automaton.CharacterRunAutomaton;
+import org.graylog.shaded.opensearch2.org.opensearch.common.regex.Regex;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class RemoteReindexAllowlist {
+
+    private final URI remoteHost;
+    private final String allowlist;
+
+    public RemoteReindexAllowlist(URI remoteHost, String requestedAllowlist) {
+        this.remoteHost = remoteHost;
+        this.allowlist = requestedAllowlist;
```

Comment:
I think I would validate and parse into the list during construction to avoid multiple method calls

Reply:
You are right, that's better. Implemented.